### PR TITLE
2.0.0-dev.2.7

### LINKED ISSUE #101: Clarify "Unsupported Extension" means the validator doesn't support the extension
Since the Babylon.js Sandbox and three.js gltf-viewer both provide validation results along with the viewer some developers are getting confused that it might be the viewers that don't support the extension. 

It might be useful to clarify in the message that "Unsupported Extension" simply means validator does not support it and therefore cannot fully validate the file.
Maybe something like: 
"message": "Cannot validate extension as it is not supported by validator: 'KHR_lights_punctual'.",

Also is there a list of extensions supported by the validator somewhere? Couldn't seem to find it.

### LINKED ISSUE #102: Add a new warning on potential primitive restart
After merging https://github.com/KhronosGroup/glTF/pull/1476.

### LINKED ISSUE #103: Add a hint on using Data URIs in GLB
See https://github.com/KhronosGroup/glTF-Blender-IO/issues/61.

### LINKED ISSUE #107: Update indexed attribute semantics check
See https://github.com/KhronosGroup/glTF/pull/1465.

### LINKED ISSUE #108: Issue an error on null value in extensions dictionary
See https://github.com/KhronosGroup/glTF-Blender-IO/issues/157.

### PR DESCRIPTION
**Pull request by lexaknyazev**
- [x] Added `ACCESSOR_INDEX_PRIMITIVE_RESTART` (fixes #102)
- [x] Added `DATA_URI_GLB` (fixes #103)
- [x] Update language for `UNSUPPORTED_EXTENSION`, publish a list of supported extensions (fixes #101)
- [x] JS API can now provide a list of supported extensions.
- [x] Updated indexed semantics validation to match the updated spec (fixes #107).
- [x] Unsupported extension objects cannot be `null` anymore (fixes #108).
- [x] Add support for `KHR_lights_punctual`
- [x] Renamed error code `NODE_ROTATION_NON_UNIT` to `ROTATION_NON_UNIT`.

### CODE REVIEW

**Comment by lexaknyazev** [comment]
Please keep it open, I'll implement lights support over this weekend.

**Comment by zellski** [comment]
I'm adding it to FBX2glTF this weekend, too, for what that might be worth.

**Comment by lexaknyazev** [comment]
This is ready for review.

File: lib/src/ext/KHR_lights_punctual/khr_lights_punctual.dart [hunk 1]
```diff
@@ -0,0 +1,265 @@
+/*
+ * # Copyright (c) 2016-2017 The Khronos Group Inc.
+ * # Copyright (c) 2016 Alexey Knyazev
```
[the remark this answers is not in the document]
Comment (lexaknyazev) [reply lib/src/ext/KHR_lights_punctual/khr_lights_punctual.dart]: This comes from a template made long ago. Probably, all files should be updated. Can we postpone that to the next release?

[the remark this answers is not in the document]
Comment (emackey) [reply lib/src/ext/KHR_lights_punctual/khr_lights_punctual.dart]: Sure.

**Comment by donmccurdy** [comment]
Commented on a couple typos, otherwise looks good! The `supportedExtensions()` method will be helpful thanks. :)

**Comment by emackey** [comment]
`gltf-vscode` got some major updates published today, including Babylon Inspector integration, and also it bundles this version of the validator now.

**Pull request merged**